financial-connections/src/main/java/com/stripe/android/financialconnections/features/error/ErrorScreen.kt

```diff
@@ -0,0 +1,156 @@
+import com.stripe.android.financialconnections.exception.InstitutionPlannedDowntimeError
+import com.stripe.android.financialconnections.exception.InstitutionUnplannedDowntimeError
+import com.stripe.android.financialconnections.exception.PartnerAuthError
+import com.stripe.android.financialconnections.features.common.FullScreenGenericLoading
+import com.stripe.android.financialconnections.features.common.InstitutionPlannedDowntimeErrorContent
+import com.stripe.android.financialconnections.features.common.InstitutionUnknownErrorContent
+import com.stripe.android.financialconnections.features.common.InstitutionUnplannedDowntimeErrorContent
+import com.stripe.android.financialconnections.features.common.UnclassifiedErrorContent
+import com.stripe.android.financialconnections.presentation.parentViewModel
+import com.stripe.android.financialconnections.ui.FinancialConnectionsPreview
+import com.stripe.android.financialconnections.ui.components.FinancialConnectionsScaffold
+import com.stripe.android.financialconnections.ui.components.FinancialConnectionsTopAppBar
+
+@Composable
+internal fun ErrorScreen() {
+    val viewModel: ErrorViewModel = mavericksViewModel()
+    val parentViewModel = parentViewModel()
+    BackHandler(true) { }
+    val payload = viewModel.collectAsState { it.payload }
+    ErrorContent(
+        payload = payload.value,
+        onManualEntryClick = viewModel::onManualEntryClick,
+        onSelectBankClick = viewModel::onSelectAnotherBank,
+        onCloseFromErrorClick = parentViewModel::onCloseFromErrorClick
+    )
+}
+
+@Composable
+private fun ErrorContent(
+    payload: Async<ErrorState.Payload>,
+    onSelectBankClick: () -> Unit,
+    onManualEntryClick: () -> Unit,
+    onCloseFromErrorClick: (Throwable) -> Unit
+) {
+    when (payload) {
+        Uninitialized,
+        is Loading -> FullScreenGenericLoading()
+
+        // Render error successfully retrieved from a previous pane
+        is Success -> ErrorContent(
+            payload().error,
+            onSelectAnotherBank = onSelectBankClick,
+            onEnterDetailsManually = onManualEntryClick,
+            onCloseFromErrorClick = onCloseFromErrorClick
+        )
+
+        // Something wrong happened while trying to retrieve the error, render the unclassified error
+        is Fail -> ErrorContent(
+            payload.error,
+            onSelectAnotherBank = onSelectBankClick,
+            onEnterDetailsManually = onManualEntryClick,
+            onCloseFromErrorClick = onCloseFromErrorClick
+        )
+    }
+}
+
+@Composable
+private fun ErrorContent(
+    error: Throwable,
+    onSelectAnotherBank: () -> Unit,
+    onEnterDetailsManually: () -> Unit,
+    onCloseFromErrorClick: (Throwable) -> Unit
+) {
+    when (error) {
+        is InstitutionPlannedDowntimeError -> FullScreenError(
+            showBack = false,
+            onCloseClick = { onCloseFromErrorClick(error) },
+            content = {
```

Comment:
That's a good idea! callbacks are the only piece that I'm not convinced about. Passing all the possible callbacks to the `Content` function and letting the per-exception implementation handle it works, but maybe it does not scale as we start adding more exceptions. 

But that's what [web does](https://git.corp.stripe.com/stripe-internal/stripe-js-v3/blob/c6b08376/src/linkedAccounts/inner/components/panes/v3/FailurePane/index.tsx#L106-L113) so it might work for us!